azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -765,24 +766,26 @@ class KustoClient(_KustoClientBase):
     # The maximum amount of connections to be able to operate in parallel
     _max_pool_size = 100
 
-    def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str]):
+    def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str], proxies: Optional[Dict[str, str]] = None):

```

Comment:
Do we have to pass it in the C'tor? I know it's optional but also an uncommon use case..

Reply:
It's the easiest way to ensure it's in propagated all the way down

Reply:
You can add it as a property and override the necessary members when the user sets it - but your call

Reply:
Ok I thought about it and I think you're right.
I changed it, let me know what you think